raiden_libs/network/matrix/client.py

```diff
@@ -175,6 +183,22 @@ def start_listener_thread(self, timeout_ms: int = 30000, exception_handler: Call
         self.should_listen = True
         self.sync_thread = gevent.spawn(self.listen_forever, timeout_ms, exception_handler)
 
+    def stop_listener_thread(self):
+        """ Kills sync_thread greenlet before joining it """
+        # when stopping, `kill` will cause the `self.api.sync` call in _sync
+        # to raise a connection error. This flag will ensure it exits gracefully then
+        self.should_listen = False
+        if self.sync_thread:
+            self.sync_thread.kill()
```

Comment:
the kill should be removed from here, otherwise the flag wont have effect.

Reply:
Yes, it will. As I explained before, this ensures these long-polling http requests at `_sync` (which can take up to 30s) need to be killed to avoid waiting this timeout to run next loop, and on `kill`, it'll also bubble up as a `Connection aborted` error wrapped in a `MatrixHttpLibError`, which will then be ignored, looped again and exited, as the flag is set to `False`.

Reply:
Okay, I misunderstood the comment.

This is getting ridiculously complicated. Exceptions are being used for control flow, the logic is tons of different projects (greenlet/gevent/requests/matrix/raiden-libs/raiden), we need to know requests wraps greenlet's kill with a connection aborted, that matrix wraps connection aborted with lib error, that raiden has a short timeout for shutdown, that raiden-libs has a long poll which has a long waiting timeout .... yada, yada.

Why don't you just increase the waiting timeout in the Raiden client?

Reply:
That may increase in 30s each test teardown, I don't think this is required, if you can keep these details in mind, and that only happens on tests (as usually your matrix server won't simply vanish, and if it does, everything else will make sure your node crashes on this)

Reply:
Reducing these timeouts will mean pounding the server. killing the only connection which will take more than a few `ms` will do what we expect without burdening even more the server with unneeded connections.

Reply:
> Reducing these timeouts will mean pounding the server. killing the only connection which will take more than a few ms will do what we expect without burdening even more the server with unneeded connections.

Are we not talking about tests? What is the problem of pounding the local server used exclusively for testing?